src/include/processor/operator/persistent/reader.h

```diff
@@ -66,8 +65,7 @@ class Reader : public PhysicalOperator {
 
     read_rows_func_t readFunc;
     init_reader_data_func_t initFunc;
-    // For parallel reading.
-    std::unique_ptr<ReaderFunctionData> readFuncData;
+    std::shared_ptr<ReaderFunctionData> readFuncData;
```

Comment:
Can we store `readFuncData` in the globalState and keep raw pointers in the operator?

Reply:
We also need to keep a smart pointer for it when reader is parquet or npy reader.